### PR TITLE
Support dynamically adding log4j2 properties via MI CLI

### PR DESCRIPTION
**Pull request by Vathsan**
Fixes https://github.com/wso2/micro-integrator/issues/1156

### CODE REVIEW

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java [hunk 1]
```diff
@@ -29,6 +29,9 @@
 import org.apache.synapse.rest.RESTConstants;
 import org.json.JSONArray;
 import org.json.JSONObject;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java]: shall we add these to bundle imports as well

Comment (Vathsan) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java]: Fixed in 8d125b0b73b6f415386fdfeb92c6212426f04832

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/pom.xml [hunk 2]
```diff
@@ -52,10 +52,12 @@
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
                         <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Private-Package>org.wso2.micro.integrator.management.apis.internal</Private-Package>
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/pom.xml]: Do we need to explicitly private package this?

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/internal/LoggingUpdaterServiceComponent.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.event.EventConstants;
+import org.wso2.micro.integrator.management.apis.Constants;
+
+/**
+ * Logging updater service component.
+ */
+@Component(name = "org.wso2.micro.integrator.management.apis", immediate = true, property = EventConstants.EVENT_TOPIC
+        + "=" + Constants.PAX_LOGGING_CONFIGURATION_TOPIC)
+public class LoggingUpdaterServiceComponent {
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/internal/LoggingUpdaterServiceComponent.java]: Can we use 'ServiceBusInitializer' without introducing a new service component?

Comment (Vathsan) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/internal/LoggingUpdaterServiceComponent.java]: Done in 86d3c947baab9b88291b959d3c35e2d03699c888

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/pom.xml (hunk 2 above)
Comment (Vathsan) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/pom.xml]: Removed the private package in 86d3c947baab9b88291b959d3c35e2d03699c888

File: components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java [hunk 4]
```diff
@@ -515,11 +515,15 @@ protected void unsetTaskService(TaskService taskService) {
...
-        log.debug("Set ConfigAdmin Service");
+        if (log.isDebugEnabled()) {
+            log.debug("Setting ConfigurationAdmin Service");
+        }
         ConfigurationHolder.getInstance().setConfigAdminService(configAdminService);
     }
 
     public void unsetConfigAdminService(ConfigurationAdmin configAdminService) {
-        log.debug("Unset ConfigAdmin Service");
+        if (log.isDebugEnabled()) {
```
Comment (Nirothipan) [inline components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java]: we don't need the if here as we are not doing any string concatenation. Same above.

Comment (Vathsan) [reply components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/ServiceBusInitializer.java]: Done

File: components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java [hunk 5]
```diff
@@ -106,4 +109,19 @@ public static ServiceAdmin getServiceAdmin(MessageContext messageContext) {
...
         return serviceAdmin;
     }
+
+    /**
+     * Method to update pax-logging configuration.
+     */
+    public static void updateLoggingConfiguration() throws IOException {
+
+        ConfigurationAdmin configurationAdmin = ConfigurationHolder.getInstance().getConfigAdminService();
+        if (configurationAdmin != null) {
```
Comment (sdkottegoda) [inline components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java]: Can this ever be null? ServiceBusInitializer will not get activated without this.

Comment (Vathsan) [reply components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java]: Fixed

**Pull request merged**